mining.go

```diff
@@ -1113,24 +1098,15 @@ func NewBlockTemplate(policy *mining.Policy, server *server, payToAddress dcruti
 	// 8. Select the one with the largest penalty ratio (highest block reward).
 	//    This block is then selected to build upon instead of the others, because
 	//    it yields the greater amount of rewards.
-	chainState.Lock()
-	prevHash := chainState.newestHash
-	nextBlockHeight := chainState.newestHeight + 1
-	poolSize := chainState.nextPoolSize
-	reqStakeDifficulty := chainState.nextStakeDifficulty
-	finalState := chainState.nextFinalState
-	winningTickets := make([]chainhash.Hash, len(chainState.winningTickets))
-	copy(winningTickets, chainState.winningTickets)
-	missedTickets := make([]chainhash.Hash, len(chainState.missedTickets))
-	copy(missedTickets, chainState.missedTickets)
-	chainState.Unlock()
-
-	chainBest := blockManager.chain.BestSnapshot()
-	if *prevHash != chainBest.Hash ||
-		nextBlockHeight-1 != chainBest.Height {
+
+	best := blockManager.chain.BestSnapshot()
+	prevHash := best.Hash
+	nextBlockHeight := best.Height + 1
+	if prevHash != best.Hash ||
```

Comment:
Does this check make sense, given that you just set `prevHash := best.Hash`? What could cause them to not be equal?
It seems that given you're now using `BestSnapshot` exclusively (vs using `chainState`) this whole test can be dropped as it doesn't make sense anymore (`bestSnapshot` should not change during its lifetime).